packages/codec/lib/interface/decoders/contract.ts

```diff
@@ -79,53 +83,101 @@ export default class ContractDecoder {
     }
   }
 
+  /**
+   * @hidden
+   */
   public async init(): Promise<void> {
     this.contractNetwork = (await this.web3.eth.net.getId()).toString();
   }
 
+  /**
+   * Constructs a contract instance decoder for a given instance of this contract.
+   * @param address The address of the contract instance decode.  If left out, it will be autodetected.
+   */
   public async forInstance(address?: string): Promise<ContractInstanceDecoder> {
     let instanceDecoder = new ContractInstanceDecoder(this, address);
     await instanceDecoder.init();
     return instanceDecoder;
   }
 
+  /**
+   * Takes a Web3 Transaction object and returns a copy of that object but with
```

Comment:
Can we define a Transaction interface internally so that can appear in the docs, also?

Reply:
Maybe we could just link to the web3 docs?  I don't know that we want to be defining that interface ourselves...

Reply:
Sounds good. I'll debate out-of-band the value of having our own interface